R/plotconfiguration-histogram.R

```diff
@@ -0,0 +1,143 @@
+#' @title HistogramPlotConfiguration
+#' @docType class
+#' @description  Plot Configuration for Histograms
+#' @export
+HistogramPlotConfiguration <- R6::R6Class(
+  "HistogramPlotConfiguration",
+  inherit = PlotConfiguration,
+
+  public = list(
+    mapData = NULL,
+    bins = NULL,
+    binWidth = NULL,
+    verticalLineFunctions = NULL,
+    verticalLineFunctionNames = NULL,
+
+    #verticalLineProperties = NULL,
+
+    initialize = function(#verticalLineProperties = tlfEnv$currentTheme$verticalLineProperties,
+      title = "Histogram",
+      subtitle = paste("Date:", format(Sys.Date(), "%y-%m-%d")),
+      xlabel = NULL,
+      ylabel = NULL,
+      watermark = tlfEnv$currentTheme$watermarkText,
+      data = NULL,
+      metaData = NULL,
+      dataMapping = NULL,
+      binWidth = NULL,
+      bins = NULL,
+      verticalLineFunctions = NULL,
+      verticalLineFunctionNames = NULL,
+      ...) {
+
+      super$initialize(...,
+        title = title,
+        subtitle = subtitle,
+        xlabel = xlabel,
+        ylabel = ylabel,
+        watermark = watermark,
+        data = data,
+        metaData = metaData,
+        dataMapping = dataMapping
+      ) #self$verticalLineProperties <- verticalLineProperties
+
+      self$binWidth = binWidth
+      self$bins = bins
+
+      self$verticalLineFunctions = verticalLineFunctions
+      #print( self$verticalLineFunctions )
+      self$verticalLineFunctionNames = verticalLineFunctionNames
+
+    },
+
+
+    addHistograms = function(plotObject,
```

Comment:
I don't like the order of thigs here. Optional parameters should come after mandatory parameters no?